### PR TITLE
Allow mappings to be named (solc-0.8.18 feature)

### PR DESCRIPTION
**Pull request by seanyoung**
Fixes https://github.com/hyperledger/solang/issues/1159

Signed-off-by: Sean Young <sean@mess.org>

### CODE REVIEW

**Comment by leekt** [comment]
I think it should also consider named key value

```
mapping(address user => uint256 timestamp) public lastUpdated
```

something like this

**Comment by seanyoung** [comment]
> I think it should also consider named value
> 
> ```
> mapping(address user => uint256 timestamp) public lastUpdated
> ```

Thanks for pointing that out - that was an oversight. Fixed now.

**Comment by seanyoung** [comment]
Note that the type contract variables was not included in the ast dot file. That is fixed now, but it does make the diff a lot bigger.

File: docs/language/types.rst [hunk 1]
```diff
@@ -391,6 +391,11 @@ Mappings are declared with ``mapping(keytype => valuetype)``, for example:
 .. include:: ../examples/mapping_type.sol
   :code: solidity
 
+Mappings may have a name for the key or the value, for example:
+``mapping(address owner => uint64 balance)``. The names are used in the metadata of
+the contract. If the mapping is public, then the accessor function will have
```
Comment (LucasSte) [inline docs/language/types.rst:396]:
```suggestion
the contract. If the mapping is public, the accessor function will have
```

**Review by LucasSte — APPROVED** [review]

File: docs/language/types.rst (hunk 1 above)
Comment (seanyoung) [reply docs/language/types.rst:396]: Thanks

**Review by xermicus — APPROVED** [review]

**Pull request merged**